source/shader/LineShader.cpp

```diff
@@ -158,6 +165,36 @@ void LineShader::Init()
 
 
 void LineShader::Draw(const Point &from, const Point &to, float width, const Color &color, bool roundCap)
+{
+	DrawGradient(from, to, width, color, color);
+}
+
+
+
+void LineShader::DrawDashed(const Point &from, const Point &to, const Point &unit, const float width,
+		const Color &color, const double dashLength, double spaceLength, bool roundCap)
+{
+	const double length = (to - from).Length();
+	const double patternLength = dashLength + spaceLength;
+	int segments = static_cast<int>(length / patternLength);
+	// If needed, scale pattern down so we can draw at least two of them over length.
+	if(segments < 2)
+	{
+		segments = 2;
+		spaceLength *= length / (segments * patternLength);
+	}
+	spaceLength /= 2.;
+	float capOffset = roundCap ? width : 0.;
+	for(int i = 0; i < segments; ++i)
+		Draw(from + unit * ((i * length) / segments + spaceLength + capOffset),
+			from + unit * (((i + 1) * length) / segments - spaceLength - capOffset),
```

Comment:
Why the parentheses?
```suggestion
		Draw(from + unit * (i * length / segments + spaceLength + capOffset),
			from + unit * ((i + 1) * length / segments - spaceLength - capOffset),
```